src/structs/lepton_format.rs

```diff
@@ -263,7 +264,11 @@ fn run_lepton_decoder_threads<R: Read + Seek, P: Send>(
     let pts = ProbabilityTablesSet::new();
     let mut qt = Vec::new();
     for i in 0..lh.jpeg_header.cmpc {
-        qt.push(QuantizationTables::new(&lh.jpeg_header, i));
+        let qtables = QuantizationTables::new(&lh.jpeg_header, i);
+        if qtables.get_quantization_table()[0] == 0 {
```

Comment:
I do not understand the zero check. Please add comments why it is the right thing to do.

Reply:
Same below